pkg/config/workloads.go

```diff
@@ -45,9 +62,53 @@ func (w *WorkloadConfig) Validate(workloadName string) error {
 	if w.ActivationAnnotation == "" {
 		return errors.Errorf("annotation shouldn't be empty for workload %q", workloadName)
 	}
+	if err := w.ValidateWorkloadAllowedAnnotations(); err != nil {
+		return err
+	}
 	return w.Resources.ValidateDefaults()
 }
 
+func (w *WorkloadConfig) ValidateWorkloadAllowedAnnotations() error {
+	disallowed, err := validateAllowedAndGenerateDisallowedAnnotations(w.AllowedAnnotations)
+	if err != nil {
+		return err
+	}
+	logrus.Debugf(
+		"Allowed annotations for workload: %v", w.AllowedAnnotations,
```

Comment:
do you want this still?  Or perhaps should it be disallowed rather than allowed?